src/constructors/adjacency.c

```diff
@@ -19,13 +19,17 @@
    02110-1301 USA
 
 */
+#include "igraph_sparsemat.h"
```

Comment:
Move this #include to the second group of includes; the usual rule is that we first `#include` the igraph header that directly "belongs to" the file, then any additional headers that we need, in a separate group.